src/Server.php

```diff
@@ -18,14 +18,26 @@ public function __construct(LoopInterface $loop)
 
     public function listen($port, $host = '127.0.0.1')
     {
-        if (strpos($host, ':') !== false) {
+        $localSocket = '';
+        if (filter_var($host, FILTER_VALIDATE_IP, FILTER_FLAG_IPV4)) {
+            $localSocket = 'tcp://' . $host . ':' . $port;
+        } elseif (filter_var($host, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6)) {
             // enclose IPv6 addresses in square brackets before appending port
-            $host = '[' . $host . ']';
+            $localSocket = 'tcp://[' . $host . ']:' . $port;
+        } elseif (preg_match('#^(unix|udg)://#', $host)) {
```

Comment:
UDG sockets are based on datagrams instead of streams of bytes. They're already part of [React's datagram component](https://github.com/reactphp/datagram).